nixos/modules/services/cluster/kubernetes/default.nix

```diff
@@ -75,29 +75,30 @@ let
 
   secret = name: "${cfg.secretsPath}/${name}.pem";
 
-  mkKubeConfigOptions = prefix: {
+  mkKubeConfigOptions = prefix: follows: {
     server = mkOption {
       description = "${prefix} kube-apiserver server address.";
       type = types.str;
+      default = if follows != null then follows.server else "";
     };
 
     caFile = mkOption {
       description = "${prefix} certificate authority file used to connect to kube-apiserver.";
       type = types.nullOr types.path;
-      default = cfg.caFile;
+      default = if follows != null then follows.caFile else cfg.caFile;
       defaultText = literalExpression "config.${opt.caFile}";
```

Comment:
I left the description as-is because it's not necessary to expose this cascading defaults mechanism to the user. The default value will still evaluate to the same expression in the end.